src/System.IO.Compression/src/System/IO/Compression/PositionPreservingWriteOnlyStreamWrapper.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.IO.Compression
+{
+    internal sealed class PositionPreservingWriteOnlyStreamWrapper : Stream
```

Comment:
Nit: I'd drop the "Wrapper" suffix.  There are a variety of streams that wrap other streams and that don't use a suffix like this, e.g. DeflateStream, BufferedStream, etc.